src/main/code.ts

```diff
@@ -5,125 +5,422 @@
-	figma.closePlugin("🛑 Selected node must be an instance of a component");
-  } else {
-	// This shows the HTML page in "ui.html".
-	figma.showUI(__html__, { width: 300, height: 366, title: "Table Builder" });
-  }
-  
-  figma.ui.onmessage = msg => {
-	console.log(msg);
-	
-	if (msg.type === 'build-table') {
-	  
-	  var textNode = getFirstChildOfTypeText(figma.currentPage.selection[0])
-	  if (textNode == null) {
-		figma.closePlugin("🛑 Selected node must have a child of type TEXT");
-	  } else {
-		var fontName: FontName = textNode.fontName as FontName
-		Promise.all([
-		  figma.loadFontAsync({family: fontName['family'], style: fontName["style"]}),
-		]).then(() => { 
-			var headerCells = createTable(
-			  figma.currentPage.selection[0] as FrameNode, 
-			  msg.wrap_in_autolayout,
-			  msg.right_align_numbers,
-			  msg.data)
-			figma.currentPage.selection = headerCells;
+if (figma.command == "generate-table") {
+	if (figma.currentPage.selection.length != 1) {
+		figma.closePlugin("🛑 A single node must be selected");
+	} else if (figma.currentPage.selection[0].type !== 'INSTANCE') {
+		figma.closePlugin("🛑 Selected node must be an instance of a component");
+	} else {
+		// This shows the HTML page in "ui.html".
+		figma.showUI(__html__, { width: 300, height: 366, title: "Table Builder" });
+	}
+
+	figma.ui.onmessage = msg => {
+		console.log(msg);
+
+		if (msg.type === 'build-table') {
+
+			var textNode = getFirstChildOfTypeText(figma.currentPage.selection[0])
+			if (textNode == null) {
+				figma.closePlugin("🛑 Selected node must have a child of type TEXT");
+			} else {
+				var fontName: FontName = textNode.fontName as FontName
+				Promise.all([
+					figma.loadFontAsync({ family: fontName['family'], style: fontName["style"] }),
+				]).then(() => {
+					var headerCells = createTable(
+						figma.currentPage.selection[0] as FrameNode,
+						msg.wrap_in_autolayout,
+						msg.right_align_numbers,
+						msg.data)
+					figma.currentPage.selection = headerCells;
+					figma.closePlugin();
+				}
+				)
+			}
+		} else {
+			// user did cancel
 			figma.closePlugin();
-		  }
-		)
+		}
+	};
+
+} else if (figma.command == "update-table") {
+	figma.showUI(__html__, { width: 300, height: 366, title: "Table Builder" });
 
-	  }
-  
+	figma.ui.onmessage = msg => {
+		//console.log(msg);
 
-	  
-	} else {
-	  // user did cancel
-	  figma.closePlugin();
-	}
-  };
-  
-  function createTable(
-	original: FrameNode, 
-	wrapInAutoLayout: boolean, 
+		if (figma.currentPage.selection.length != 1) {
+			figma.closePlugin("🛑 A single node must be selected");
+		} else if (msg.type === 'build-table') {
+			var firstTextNode = getFirstChildOfTypeText(figma.currentPage.selection[0])
+			if (firstTextNode == null) {
+				// No text to update was found
+				figma.closePlugin("🛑 Your selected node needs to have text");
+			} else {
+				var fontNames: FontName[] = [firstTextNode.fontName as FontName]
```

Comment:
We're probably only interested in all _distinguished_ font types, correct? If that's the case, please use a [Set](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Set) instead of an array.

Reply:
I will try to use a set in my next commit. Thanks for the suggestion!